google_bigquery_writer/app.py

```diff
@@ -17,17 +18,67 @@ def __init__(self):
         self.cfg = docker.Config(self.data_dir)
         self.writer = None
 
-    def get_credentials(self):
-        oauthapi_data = self.cfg.get_oauthapi_data()
-        if oauthapi_data == {}:
+    def validate_credentials(self):
+        parameters = self.cfg.get_parameters()
+        if (
+                parameters.get('service_account')
+        ):
+            private_key = parameters.get('service_account').get('#private_key')
+            if private_key == '' or private_key is None:
+                raise UserException('Service account private key missing.')
+
+            client_email = parameters.get('service_account').get('client_email')
+            if client_email == '' or client_email is None:
+                raise UserException('Service account client email missing.')
+
+            token_uri = parameters.get('service_account').get('token_uri')
+            if token_uri == '' or token_uri is None:
+                raise UserException('Service account token URI missing.')
+
+            project_id = parameters.get('service_account').get('project_id')
+            if project_id == '' or project_id is None:
+                raise UserException('Service account project id missing.')
+
+        if (
+                self.cfg.get_oauthapi_data() == {} and
+                not parameters.get('service_account')
+        ):
             raise UserException('Authorization missing.')
-        return google.oauth2.credentials.Credentials(
-            oauthapi_data.get('access_token'),
-            token_uri='https://accounts.google.com/o/oauth2/token',
-            client_id=self.cfg.get_oauthapi_appkey(),
-            client_secret=self.cfg.get_oauthapi_appsecret(),
-            refresh_token=oauthapi_data.get('refresh_token')
-        )
+
+    def get_credentials(self):
+        parameters = self.cfg.get_parameters()
+        if (
+                parameters.get('service_account')
+        ):
+            private_key = parameters.get('service_account').get('#private_key')
+            client_email = parameters.get('service_account').get('client_email')
+            token_uri = parameters.get('service_account').get('token_uri')
+
+            # replace all escaped newline characters
+            service_account_info = {
+                'private_key': private_key.replace('\\n', '\n'),
```

Comment:
ten replace je tam podla mna zbytocny.

Reply:
taky si myslím! ale nebyl jsem si jistej, v jaký podobě to tam jsme schopný dodat z UI

Reply:
vyzkoušel jsem to bez něj a vypadá to gut, pushuju fix, dík :-)